### ISSUE TITLE
Big line support lost between 0.2.2 and 0.2.3

### ISSUE THREAD

**Issue by wolfgang42** [nacholibre/node-readlines#6]
I've just upgraded from `n-readlines@0.2.2` and discovered that my program no longer reads its logfiles correctly. (It's one line of JSON per event, and events can have a lot of data.)

Comparing the tarballs for 0.2.2 and 0.2.3 (the commits for 0.2.2 are missing from the repository?), I see that the code seems to have been massively rewritten, and the tests and code for long lines have been lost in the process.

Was it intentional to drop long line support, or was this an oversight?

**Comment by jagtapAmit** [comment]
Does the issue get solved if we switch to older versions ?

**Comment by nacholibre** [comment]
This should be fixed in version 0.2.5, please check.

**Comment by jagtapAmit** [comment]
I tried it but for some reason the line is not read completely and ends up picking incomplete json models, funny thing is it was working  few days ago.

**Comment by nacholibre** [comment]
I didn't change a thing in the last month or two. Can you post sample code?

**Comment by jagtapAmit** [comment]
```
 var liner = new lineByLine(source_path);
 var target_model = parseLine(line);

function parseLine (line) {
        try {
            line = line.toString('utf8');
            line = line.replace(/(^[,\s]+)|([,\s]+$)/g, '');
            line = line.replace(/^"(.*)"$/, '$1');
            var target_model = JSON.parse(line);
            return target_model;
        }
        catch (err) {
            return false;
        }
    };
```

And the Json I get in 'line' variable is as below ,dont know why as there is no eol before it..
"ritalStatus":"Widowed","primaryNeed":"Absent parenting"},"

**Comment by nacholibre** [comment]
Can you create simple file and try to reproduce the actual error in the parsing?

Line by line reader looks for a new line character "\n" and then returns that line, maybe you have new line before the end of the JSON?

**Comment by jagtapAmit** [comment]
ok

**Comment by inossidabile** [comment]
I confirm the issue exists. The long lines are evaluated as multiple lines. Got updated to `0.2.5` and things broke.

**Comment by nacholibre** [comment]
Can you post example, If I can reproduce it, I'll fix it.

Regards

**Comment by inossidabile** [comment]
https://dl.dropboxusercontent.com/u/38873758/push.raw.json

**Comment by inossidabile** [comment]
Downgraded to `0.2.2` and it works again. FYI.

**Comment by nacholibre** [comment]
Okay, I was able to reproduce the bug and made a fix, can you please try with version 0.2.7 and tell me does it work?
